File: 206-reverse-linked-list/206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* l) {
         ListNode* q=NULL;
        ListNode* r=NULL;
        
        while(l!=NULL){
            q=r;
            r=l;
            l=l->next;
            r->next=q;
        }
       return r;
    
    }
};

File: 206-reverse-linked-list/206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode p=head;
        ListNode r=null;
        ListNode t=null;
        while(p!=null){
            t=p.next;
            p.next=r;
            r=p;
            p=t;
        }
        return r;
    }
}